### PR TITLE
Improved performance of table key generation

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

File: keys/keys.go [hunk 1]
```diff
@@ -63,7 +63,11 @@ func MakeNamespaceMetadataKey(namespace string) proto.Key {
 
 // MakeTableMetadataKey returns the key for the table in namespaceID.
 func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
-	return MakeKey(TableMetadataPrefix, encoding.EncodeUvarint(nil, uint64(namespaceID)), proto.Key(tableName))
+	k := make([]byte, 0, len(TableMetadataPrefix)+10+len(tableName))
+	k = append(k, TableMetadataPrefix...)
+	k = encoding.EncodeUvarint(k, uint64(namespaceID))
+	k = append(k, []byte(tableName)...)
```
Comment (petermattis) [inline keys/keys.go]: You shouldn't need the `[]byte` cast around `tableName`: http://play.golang.org/p/Wc6l0BFBtr

FYI, casts from `[]byte` to `string` and vice-versa require an allocation.

**Comment by petermattis** [comment]
LGTM

**Comment by bdarnell** [comment]
The `-` may not be strictly necessary with ordered encoding but it does improve readability, and gives us more flexibility in assigning future prefixes (if we want more than one key per namespace in the future we'd be able to use `ns-` and `nsfoo-` instead of coming up with some prefix for the new keys that does not begin with `ns`). If we're concerned about saving a byte here we should apply that logic to more than just separators: if it's worth truncating `tbl-` to `tbl` it's probably worth going all the way to `tb`.

**Comment by petermattis** [comment]
My comment about a separator not being needed for ordered code was in reference to using `-` in between ordered code encoded components of the key. I'm fine with a separator at the end of our system prefixes, though if we use one we should be consistent.

**Comment by bdarnell** [comment]
Ah, I see that now in an older commit. It's true that we don't currently use separators in most of our keys (with the exception of `status-nodes-` and `status-stores-`) and we should be consistent. I'd be in favor of adding a trailing `-` to all of the key constants that are used as prefixes, but that would be a separate change.

**Comment by vivekmenezes** [comment]
Saving bytes is not a goal here. And making it human readable is also not a
goal. I'll add the - back and only leave the optimization in

**Pull request merged**